src/bin/miri.rs

```diff
@@ -211,6 +219,10 @@ impl rustc_driver::Callbacks for MiriCompilerCalls {
                 if return_code != rustc_driver::EXIT_SUCCESS {
                     eprintln!("FAILING SEED: {seed}");
                     if !many_seeds.keep_going {
+                        // drop the tracing guard before exiting, so tracing calls are flushed correctly
+                        if let Ok(mut lock) = tracing_guard.try_lock() {
+                            let _guard_being_dropped = (*lock).take();
```

Comment:
An explicit call to `drop` seems more clear here.

But it's not great tha we apparently have to remember to do this in multiple places...

Reply:
Yeah I also don't like that much, but it felt the safest option. An alternative would be to call `libc::atexit()` or to use a crate like https://crates.io/crates/ctor, though both options involve `unsafe`.

Reply:
Here's an idea: move everything in `fn after_analysis` that is after `init_late_loggers` into a separate inherent method on `MiriCompilerCalls` that returns an `ExitCode`. Have the guard inside that separate method so it gets dropped when the method returns, and then there's just a single `exit` call in `after_analysis` afterwards.

Reply:
I thought of that but `exit()` is also used inside the `par_for_each_in` to exit immediately as soon as anything fails (when `many_seeds.keep_going` is false) and afaik there is no easy and safe way to stop executing threads in Rust

Reply:
Oh, I had extracted the `exit()` out of the ifs, but ok now I made a closure